### ISSUE TITLE
Validator permits declarations in places it should not

### ISSUE THREAD

**Issue by bakkot** [shapesecurity/shift-validator-js#33]
For example, all of the following are not valid:

``` js
if (null) let a; // though `var` would be legal in place of `let`.
```

``` js
while (null) class a {}
```

``` js
a : function* b(){}
```

(Neither are permutations of the above.)

**Comment by bakkot** [comment]
Note that function declarations are never legal as the sole statement of a loop, but outside of strict mode are legal as the sole statement of a LabelledStatement or IfStatement.

**Comment by bakkot** [comment]
This is fixed in 3.0.0.
